rfcs/0138-rfc-repos.md

```diff
@@ -0,0 +1,94 @@
+---
+feature: rfc-repos
+start-date: 2022-01-10
+author: Silvan Mosberger
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: @winterqt, @lheckemann 
+shepherd-leader: @lheckemann 
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Develop and discuss RFCs in repositories instead of a pull request.
+
+# Motivation
+[motivation]: #motivation
+
+RFC discussions are currently held in single pull requests on the [rfcs](https://github.com/NixOS/rfcs/pulls) repository.
+Due to GitHub's limitations on pull requests, this quickly becomes unmanageable.
+The most prominent example of this is [the Flakes RFC](https://github.com/NixOS/rfcs/pull/49) with over 500 comments.
+
+# Detailed design
+[design]: #detailed-design
+
+The RFC process is changed to use repositories to develop and review RFCs instead.
+To create a new RFC, an issue on the [rfcs](https://github.com/NixOS/rfcs) repository is opened.
+It should contain the following contents describing the process:
+
+> This RFC is developed as the README.md of this repository: https://github.com/$owner/$repo
+>
+> To review the RFC
+> - Open https://github.com/$owner/$repo/blob/master/README.md?plain=1
+> - Select the lines you wish to comment on
+> - Click the "..." menu on the left
+> - Select "Reference in new issue" to open a new issue commenting on those lines
+>
+> To suggest changes, open a PR against the repository.
+>
+> All issues and pull requests must be resolved before the FCP can be initiated and completed.
+>
+> This issue may only be used for RFC meta-discussions, such as shepherd nominations, FCP periods, meeting schedules, etc.
+
+The FCP must be initiated on a specific commit of the repository.
+When the FCP passes, the repositories contents are committed to the [rfcs](https://github.com/NixOS/rfcs) repository by the [RFC steering committee](https://github.com/NixOS/rfcs#rfc-steering-committee) with a commit that closes the original issue.
+
+## Workflow to get RFC updates
+To get updates for RFC's, instead of subscribing to the PR, one has to watch the repository.
+
+## Repository transfer
+
+Once the RFC is merged, the repository has to be transferred to the NixOS organization under https://github.com/NixOS/rfc-NUMBER. This is to ensure the discussions aren't lost in the future. The repository will then be archived.
```

Comment:
I think it should be done in a more centralized form.

This is merely an archive of discussions. Then a possible design is

1. Create the repo `https://github.com/NixOS/rfc-archives` 
2. For each merged RFC:
   1. Create a branch `https://github.com/NixOS/rfc-archives/tree/RFC-<number>`
   2. Save (by using some git magic I don't remember perfectly now) the discussion
   3. Update `MERGED RFCS` section on README.MD (including a link to the branch created above) 
   4. Profit!

Reply:
> 2\. Save (by using some git magic I don't remember perfectly now) the discussion

The discussion happens in GitHub issues and pull requests though, it's not attached to Git itself. And while issues can be moved to separate repositories, moving discussions from separate RFCs to the same repository would create a very messy issue list.